client/v2/client.go

```diff
@@ -534,7 +548,7 @@ func (c *client) Query(q Query) (*Response, error) {
 				break
 			}
 		}
-	} else {
+	} else if strings.Contains(resp.Header.Get("Content-Type"), "application/json") {
```

Comment:
I'm not sure about needing to separate out unmarshaling a body that comes from a response with an `application/json` content type, and dealing with a body that comes from a response with a different content type.

We could just attempt to unmarshal the body into `dec`, and if there is an error then return that, along with the body.

Reply:
The reasoning behind separating it out is two-fold:

1. downstream error pages (maybe your proxy does auth and throws a 403 non-json response) gets handled without an explicit 500+ status code check
2. saves wasting time dropping into the json decode machinery and allows us to fail early _and_ keep the body reader content - dropping into the decoder will read from the body reader into a private buffer inside the decoder we can't access, so we can't return any error messages it may contain.

Reply:
Sounds reasonable 👍

Reply:
This check should be for the response as a whole. A chunked response should have this check too.

You also don't want to use `strings.Contains` here. You will want to use this: https://godoc.org/mime#ParseMediaType

So something like:

```
var contentType string
if header := resp.Header.Get("Content-Type"); header != "" {
    contentType, _, _ = mime.ParseMediaType(header)
}

if contentType != "application/json" {
    // the code you added
}

// the normal code that was already there
```

Reply:
Ahh yes, good point. I'll fix that shortly and resubmit.